src/App.tsx

```diff
@@ -8,37 +8,32 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { RootState } from './app/store';
+import { useSelector } from 'react-redux';
+import { useAppDispatch } from './app/hooks';
+import { getPostsAsync, postsSliceActions } from './features/postsSlice';
+import { SelecterPostState, selectPost } from './features/selectedPostSlice';
+import { AuthorState } from './features/authorSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const posts = useSelector((state: RootState) => state.posts.data);
+  const loaded = useSelector((state: RootState) => state.posts.loading);
+  const hasError = useSelector((state: RootState) => state.posts.error);
+  const author = useSelector((state: RootState) => state.author) as AuthorState;
+  const selectedPost = useSelector(
+    (state: RootState) => state.selectedPost,
+  ) as SelecterPostState;
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
     // we clear the post when an author is changed
     // not to confuse the user
-    setSelectedPost(null);
+    dispatch(selectPost(null));
 
-    if (author) {
-      loadUserPosts(author.id);
+    if (author && author !== null && author !== undefined && author.id) {
+      dispatch(getPostsAsync(author.id));
     } else {
-      setPosts([]);
+      dispatch(postsSliceActions.clearPosts());
     }
   }, [author]);
```

Comment:
Fix such cases
![2024-04-11_22h50_49](https://github.com/mate-academy/react_redux-list-of-posts/assets/61904995/a2fa1dd6-dda0-4daa-a2c4-3f7728c80576)